addons/point_of_sale/models/pos_session.py

```diff
@@ -408,6 +410,14 @@ def _create_account_move(self):
                         stock_expense[exp_key] = self._update_amounts(stock_expense[exp_key], {'amount': amount}, move.picking_id.date)
                         stock_output[out_key] = self._update_amounts(stock_output[out_key], {'amount': amount}, move.picking_id.date)
 
+            # Setting current partner's customer_rank to 1 if necessary
+            try:
+                with self.env.cr.savepoint():
+                    self.env.cr.execute("UPDATE res_partner SET customer_rank=1 WHERE ID=%s AND customer_rank=0", (order.partner_id.id,))
+                    self.env.cache.remove(order.partner_id, order.partner_id._fields['customer_rank'])
+            except psycopg2.DatabaseError as e:
+                raise e
```

Comment:
```suggestion
 if e.pgcode == '55P03':
                _logger.debug('Another transaction already locked partner rows. Cannot update partner ranks.')
                    continue
                else:
                    raise e
```
Don't block the user if the partner is lock

Reply:
updated accordingly